pkg/results/results.go

```diff
@@ -105,9 +99,13 @@ func NewResultCollectionFromRegoResultSet(rs *rego.ResultSet) (*ResultCollection
 
 	rc := make(ResultCollection, 0, len(keys))
 	for _, k := range keys {
+		value, ok := values[k].([]string)
```

Comment:
This breaks backward compatibility with the existing packages.

I think it is possible to add the feature while keeping b/c:

If order to do so, I'd make `NewResultCollectionFromRegoResultSet` attempt to read the value as a `bool` (the behaviour we are trying to deprecate). If it reads a bool successfully, print a "this is deprecated bleh bleh bleh" message and build the Result object creating a `[]string` that would be either `[]string{}` if the bool was `true` (previously meaning success), or `[]string{"missing context"}` (or something like that) in case the bool was `false`.

Reply:
I've had a go at this in d116daf9015a14da21ab7a00cc72ab205aacab3b - wdyt? 

I also opted to leave out a `Success` bool for now (as per https://github.com/jetstack/preflight/pull/38/files/d9e42e8d1fa65447295e4f156526fadf08d11bf3#r365223255)